src/lambda_function.py

```diff
@@ -15,136 +15,186 @@
 logger = logging.getLogger()
 logger.setLevel(logging.INFO)
 
+
 @xray_recorder.capture('lambda_handler')
 def lambda_handler(event, context):
-
-    ## event must be a dict with a url key, and context can be nothing:
-    ## lambda_handler({"url":"http://bbc.co.uk"}, "")
+    """
+    event must be a dict with a url key, and context can be nothing:
+    lambda_handler({"url":"http://bbc.co.uk"}, "")
+    """
 
     subsegment = xray_recorder.begin_subsegment('lambda_function: check URL')
     logger.info(f'LambdaFunction: Checking we have a URL...')
     try:
         url = event['url']
     except KeyError:
         logger.info(f'LambdaFunction: URL is not present.')
-        return {"error" : "No URL provided"}
+        return {"error": "No URL provided"}
 
     xray_recorder.end_subsegment()
 
-    #### Adding "https://" to the URL if not present
+    """
```

Comment:
This was a single line comment and not a docstring so it's fine to use `#` It looks a bit odd to have it as a string comment. But maybe you could have just begun with a single `#`